### PR TITLE
Add external store enabled property for publisher settings API. Add last updated time for published external stores. 

### PR DESCRIPTION
**Pull request by dushaniw**
This PR adds several improvements to External Store publishing in Publisher backend. 

- Add externalStoreEnabled property to /settings API
- Add APIExternalStoreDTO to send last updated time of external store when getting published store list for the API.
-Add LAST_UPDATED_TIME column to AM_EXTERNAL_STORES table.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/db2.sql [hunk 1]
```diff
@@ -1896,6 +1896,7 @@ CREATE TABLE AM_EXTERNAL_STORES (
             STORE_DISPLAY_NAME VARCHAR(255) NOT NULL,
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/db2.sql]: Seems we have introduced some modifications to the schema. So this has a migration impact right? So we have to track this.

Comment (dushaniw) [reply features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/db2.sql]: Yes this has an impact. Created https://github.com/wso2/product-apim/issues/5505 to track the schema migration changes.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 2]
```diff
@@ -2343,6 +2327,64 @@ public static String createEndpoint(String endpointUrl, Registry registry) throw
...
+                    .getGovernanceSystemRegistry(tenantId);
+            if (registry.resourceExists(APIConstants.EXTERNAL_API_STORES_LOCATION)) {
+                Resource resource = registry.get(APIConstants.EXTERNAL_API_STORES_LOCATION);
+                String content = new String((byte[]) resource.getContent(), Charset.defaultCharset());
+                OMElement element = AXIOMUtil.stringToOM(content);
+                apiStoreIterator = element.getChildrenWithLocalName("ExternalAPIStore");
+            }
+        } catch (RegistryException e) {
+            String msg = "Error while retrieving External Stores Configuration from registry";
+            log.error(msg, e);
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: do we need to throw and log same exception?

Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: The thrown exception is not logged in server from any of the calling methods, hence we need to log it right?

Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Found that the error is logged in GlobalThrowableMapper from [1] and [2].

[1] https://github.com/wso2/carbon-apimgt/blob/5fef47dff0a057eaaa88993ada0497db3db7286d/components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/exception/GlobalThrowableMapper.java#L168

[2]https://github.com/wso2/carbon-apimgt/blob/5fef47dff0a057eaaa88993ada0497db3db7286d/components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/exception/GlobalThrowableMapper.java#L179

Hence removed the additional error logs from APIUtil.

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/db2.sql (hunk 1 above)
Comment (harsha89) [reply features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/db2.sql]: :+1:

**Pull request merged**